drivers/net/can/usb/gs_usb.c

```diff
@@ -408,85 +511,30 @@ static void gs_usb_receive_bulk_callback(struct urb *urb)
 
 	dev = usbcan->canch[hf->channel];
 
-	netdev = dev->netdev;
-	stats = &netdev->stats;
-
-	if (!netif_device_present(netdev))
+	if (!netif_device_present(dev->netdev))
 		return;
 
-	if (hf->echo_id == -1) { /* normal rx */
-		if (hf->flags & GS_CAN_FLAG_FD) {
-			skb = alloc_canfd_skb(dev->netdev, &cfd);
-			if (!skb)
-				return;
-
-			cfd->can_id = le32_to_cpu(hf->can_id);
-			cfd->len = can_fd_dlc2len(hf->can_dlc);
-			if (hf->flags & GS_CAN_FLAG_BRS)
-				cfd->flags |= CANFD_BRS;
-			if (hf->flags & GS_CAN_FLAG_ESI)
-				cfd->flags |= CANFD_ESI;
-
-			memcpy(cfd->data, hf->canfd->data, cfd->len);
-		} else {
-			skb = alloc_can_skb(dev->netdev, &cf);
-			if (!skb)
-				return;
-
-			cf->can_id = le32_to_cpu(hf->can_id);
-			can_frame_set_cc_len(cf, hf->can_dlc, dev->can.ctrlmode);
-
-			memcpy(cf->data, hf->classic_can->data, 8);
-
-			/* ERROR frames tell us information about the controller */
-			if (le32_to_cpu(hf->can_id) & CAN_ERR_FLAG)
-				gs_update_state(dev, cf);
-		}
+	if (hf->echo_id == 0xFFFFFFFF) { /* normal rx */
+		if (gs_usb_receive_can_normal_rx(dev, hf))
+			return;
 
-		netdev->stats.rx_packets++;
-		netdev->stats.rx_bytes += hf->can_dlc;
+	} else if (hf->echo_id == 0xFFFFFFFE) { /* Terminating resistor state */
```

Comment:
Please don't abuse the `echo_id`. Please implement `GS_USB_BREQ_GET_TERMINATION`.

Reply:
Fair enough; This part doesn't work anyways.